### ISSUE TITLE
Permission issue adding jobs with custom role

### ISSUE THREAD

**Issue by ohiowebpro** [Automattic/WP-Job-Manager#2673]
Permission issue adding jobs with custom role

**Describe the bug**
We are having an issue with adding a job with a custom role. We have a custom role called Job Poster and that role has all the capabilities added, that WP Job Manager creates using a role editor plugin. Now after an update, the user with that role cannot add a new job. They can still edit and delete. Here is the troubleshooting I have done:

    Disabled all plugins except WP Job Manager and Role editor plugin. Same issue.
    Tried a different role editor plugin and made sure capabilities are added. Same issue.
    Switched to the 2024 theme. Same issue.

**To Reproduce**
1. install WP Job Manager and a role editor plugin
2. set up a custom role with all the job manager capabilities
3. log in as a user with that role and try to add a job. You should get a permission issue.

**Screenshots**
![screenshot of admin](https://imgur.com/gSJnH63)
![screenshot of error](https://imgur.com/WzHGUL9)

**Expected behavior**
To be able to add a job rather than get the permission issue warning

**Isolating the problem (mark completed items with an [x]):**
- [X ] I have deactivated other plugins and confirmed this bug occurs when only WP Job Manager plugin is active.
- [X ] This bug happens with a default WordPress theme active.
- [X ] I can reproduce this bug consistently using the steps above.

**WordPress Environment**
- WordPress Version:  6.4.2
- WP Job Manager Version: 2.1.1
- PHP Version: 8.0.28
- Other important details:

**Comment by yscik** [comment]
Linking in the support forum threads too:
https://wordpress.org/support/topic/changing-user-role-permissions/
https://wordpress.org/support/topic/permission-issue-adding-jobs-with-custom-role/
https://wordpress.org/support/topic/job-posting-required-full-access-to-all-posts-and-pages/

Seems like this is caused by some unintentional change introduced in the 2.0.0 release.

**Comment by roman-tikonov** [comment]
Hey folks, could you please let us know when do you plan to release this fix, and what is a workaround that we could use until then?

**Comment by T19M-Scott** [comment]
Hi Roman,

Like you, I am awaiting an official plugin update release.  However, I did
the following as a workaround.  I downloaded the zip file that was approved
as a fix in GitHub and installed it on my site.  If you visit the "Fixed"
Issue #2689 (link below), you'll see about halfway down where there is an
option to "Download plugin zip" under the heading "Plugin Build for 3a72581"

https://github.com/Automattic/WP-Job-Manager/pull/2689

That is essentially a zip file of the plugin with the fix in place - they
just haven't released it yet for some reason (they are probably adding some
other updates to it as well).
